// simple-react-form-bootstrap/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ObjectField = exports.ArrayField = exports.CheckboxesField = exports.RadioField = exports.AutocompleteField = exports.DatetimeField = exports.TagsField = exports.SelectField = exports.ColorField = exports.PasswordField = exports.EmailField = exports.DateField = exports.NumberField = exports.TextareaField = exports.TextField = undefined;

var _string = require('./fields/string.js');

var _select = require('./fields/select.js');

var _tags = require('./fields/tags.js');

var _datetime = require('./fields/datetime.js');

var _autocomplete = require('./fields/autocomplete.js');

var _radio = require('./fields/radio.js');

var _radio2 = _interopRequireDefault(_radio);

var _checkboxes = require('./fields/checkboxes.js');

var _checkboxes2 = _interopRequireDefault(_checkboxes);

var _array = require('./fields/array.js');

var _object = require('./fields/object.js');

var _simpleReactForm = require('simple-react-form');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var registerTypes = function registerTypes() {
  (0, _simpleReactForm.registerType)({
    type: 'string',
    component: _string.TextField
  });
  (0, _simpleReactForm.registerType)({
    type: 'text',
    component: _string.TextField
  });
  (0, _simpleReactForm.registerType)({
    type: 'textarea',
    component: _string.TextareaField
  });
  (0, _simpleReactForm.registerType)({
    type: 'number',
    component: _string.NumberField
  });
  (0, _simpleReactForm.registerType)({
    type: 'date',
    component: _string.DateField
  });
  (0, _simpleReactForm.registerType)({
    type: 'email',
    component: _string.EmailField
  });
  (0, _simpleReactForm.registerType)({
    type: 'password',
    component: _string.PasswordField
  });
  (0, _simpleReactForm.registerType)({
    type: 'color',
    component: _string.ColorField
  });
  (0, _simpleReactForm.registerType)({
    type: 'autocomplete',
    component: _autocomplete.AutocompleteField,
    allowedTypes: [String, [String]]
  });
  (0, _simpleReactForm.registerType)({
    type: 'tags',
    component: _tags.TagsField
  });
  (0, _simpleReactForm.registerType)({
    type: 'string-array',
    component: _tags.TagsField
  });
  (0, _simpleReactForm.registerType)({
    type: 'array',
    component: _array.ArrayField
  });
  (0, _simpleReactForm.registerType)({
    type: 'object',
    component: _object.ObjectField
  });
  (0, _simpleReactForm.registerType)({
    type: 'select',
    component: _select.SelectField,
    allowedTypes: [String, Number]
  });
  (0, _simpleReactForm.registerType)({
    type: 'datetime',
    component: _datetime.DatetimeField,
    allowedTypes: [Date, String]
  });
  (0, _simpleReactForm.registerType)({
    type: 'radio',
    component: _radio2.default
  });
  (0, _simpleReactForm.registerType)({
    type: 'multiple-checkbox',
    component: _checkboxes2.default
  });
};
exports.default = registerTypes;
exports.TextField = _string.TextField;
exports.TextareaField = _string.TextareaField;
exports.NumberField = _string.NumberField;
exports.DateField = _string.DateField;
exports.EmailField = _string.EmailField;
exports.PasswordField = _string.PasswordField;
exports.ColorField = _string.ColorField;
exports.SelectField = _select.SelectField;
exports.TagsField = _tags.TagsField;
exports.DatetimeField = _datetime.DatetimeField;
exports.AutocompleteField = _autocomplete.AutocompleteField;
exports.RadioField = _radio2.default;
exports.CheckboxesField = _checkboxes2.default;
exports.ArrayField = _array.ArrayField;
exports.ObjectField = _object.ObjectField;
